*: replace grpc clients with rpc adapters

cthumuluru-crdb opened the pull request:
gRPC and DRPC clients have different signatures. To unify them, we created a common interface for both types of clients, and replaced the gRPC and DRPC clients with adapters.

Epic: CRDB-48923
Informs: #148353
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/147950)
<!-- Reviewable:end -->

cthumuluru-crdb committed:
keyvisualizer: replace `KeyVisualizerClient` client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
blobs: replace `BlobClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
server,upgrade: replace `MigrationClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
util: replace `TracingClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
kv: replace `StoreLivenessClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
kv,server: replace `SideTransportClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
kv,server: replace `MultiRaftClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
kv: replace `PerStoreClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
server,sql,testutils: replace `DistSQLClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
*: replace `TimeSeriesClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
rpc: replace `HeartbeatClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
*: replace `AdminClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
cli,server,testutils: replace `InitClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
*: replace `StatusClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
ccl: replace `DirectoryClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
gossip,server: replace `GossipClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Informs:#148353
Release note: none

cthumuluru-crdb committed:
kv,server: replace `PerReplicaClient` RPC client with adapter

gRPC and DRPC clients have different signatures. To unify them, a common
interface was created for both types of clients, and replaced the gRPC
and DRPC clients with adapters.

Epic: CRDB-48923
Fixes:#148353
Release note: none

cthumuluru-crdb commented:
bors r+
